### DISCUSSION TITLE
[Mobile] C# Nuget for Android

### DISCUSSION THREAD

**Discussion by GuiOliv** [microsoft/onnxruntime-genai#665] [Support for Targets (OS / EPs / Hardware)]
I am writting an app using MAUI Blazor and I'm trying to load a mocal locally on my phone. However, everytime it hits the line var model = new Model(modelPath);, it throws the exception System.DllNotFoundException "onnxruntime-genai".

I have tried copying dll files multiple times, using them, installing the package on multiple projects but nothing worked.

To reproduce
To reproduce, you may try to load a local model by copying it to your phone and using

       string modelPath = "/storage/emulated/0/onnx/cpu_and_mobile/cpu-int4-rtn-block-32";

Then simply load the model with the line var model = new Model(modelPath);, all this while debugging with your phone via USB.

Visual Studio Community Version 17.10.0

Urgency
This is a personal project so not of much urgency but I would really appreciate if I could have any help.

Platform
Android

OS Version
14

ONNX Runtime Installation
Built from Source

Compiler Version (if 'Built from Source')
Debug / Any CPU

Package Name (if 'Released Package')
Microsoft.ML.OnnxRuntime

ONNX Runtime Version or Commit ID
Newest

ONNX Runtime API
C#

Architecture
Other / Unknown

Execution Provider
Default CPU

**Comment by BrainSlugs83** [comment]
Does it tell you which DLL it can't find?

I'm guessing you will need a different backend NuGet package for Android, but I'm not sure which one.  I do see that `Microsoft.ML.OnnxRuntime.Managed` has some Android and iOS targets though...

**Comment by baijumeswani** [comment]
We do not have a nuget package that supports Android yet.

Having said that, we [recently added Java bindings](https://github.com/microsoft/onnxruntime-genai/pull/461) and support for building for android.

Would that work for you? We can consider adding nuget support for Android in the near future.

**Comment by GuiOliv** [comment]
> We do not have a nuget package that supports Android yet.
> 
> Having said that, we [recently added Java bindings](https://github.com/microsoft/onnxruntime-genai/pull/461) and support for building for android.
> 
> Would that work for you? We can consider adding nuget support for Android in the near future.

Considering I am using C#, it wouldn't be really helpful.

Would love to see nuget support for Android tho.

Thank you for your answer @baijumeswani

**Comment by ggt-k** [comment]
Has this been even fixed? Whats the point in closing the original issue and conversion to a discussion?

**Comment by skyline75489** [comment]
@ggt-k The PR related is pending review: https://github.com/microsoft/onnxruntime-genai/pull/923 Hopefully we can have Android NuGet support in the next release.
